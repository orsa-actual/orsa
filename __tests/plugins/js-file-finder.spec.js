const fs = require('fs');
const glob = require('glob');
const jsFileFinder = require('../../src/plugins/js-file-finder');

jest.mock('fs');
jest.mock('glob');

describe('Javascript Package Indexer', () => {
  it('should index', async () => {
    glob.sync.mockImplementation(() => (['hello']));
    fs.readFileSync.mockImplementation(() => `import React from 'react';

/**
 * HaveSomeClass
 * @description A class
 */
class HaveSomeClass {
  foobar() {
  }
}

/**
 * App
 * @description Test file
 */
const App = () => (<div>Hello</div>);

export default App;`);
    let info = null;
    await jsFileFinder({}, {
      store: {
        query: () => ({
          data: {
            projects: [
              {
                id: '1',
                name: 'bar',
              },
            ],
          },
        }),
        getById: () => ({
          transient: {
            path: 'bar',
          },
        }),
        update: (inf) => {
          info = inf;
        },
      },
      logger: {
        log: () => {},
        error: () => {},
      },
    });
    expect(info).toMatchSnapshot();
  });
});
