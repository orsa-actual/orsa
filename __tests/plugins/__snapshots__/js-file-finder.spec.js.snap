// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Javascript Package Indexer should index 1`] = `
Object {
  "features": Array [
    Object {
      "end": 1,
      "from": "react",
      "id": "3d2c95a8dc27af0bc7326d0e161156b0",
      "keys": Array [
        "React",
      ],
      "parentId": "8936d8c0447672f0ac71feaaaff306f8",
      "snippet": "import React from 'react';",
      "start": 1,
      "type": "import",
    },
    Object {
      "end": 10,
      "id": "03de8ec7c7e5751e0c0007d513667394",
      "jsDoc": Object {
        "description": "HaveSomeClass",
        "tags": Array [
          Object {
            "description": "A class",
            "title": "description",
          },
        ],
      },
      "metadata": Object {},
      "methods": Array [
        Object {
          "end": 9,
          "id": "ce0969a2d00abafcd94cb233ad54024e",
          "jsDoc": Object {
            "description": "",
            "tags": Array [],
          },
          "kind": "method",
          "name": "foobar",
          "params": Array [],
          "parentId": "03de8ec7c7e5751e0c0007d513667394",
          "snippet": "  foobar() {
  }",
          "start": 8,
          "type": "method",
        },
      ],
      "name": "HaveSomeClass",
      "parentId": "8936d8c0447672f0ac71feaaaff306f8",
      "snippet": "class HaveSomeClass {
  foobar() {
  }
}",
      "start": 7,
      "superClass": undefined,
      "type": "class",
    },
    Object {
      "end": 16,
      "id": "36e0b0f0b99ab38d9f1748a7923782a7",
      "jsDoc": Object {
        "description": "",
        "tags": Array [],
      },
      "metadata": Object {
        "react": true,
        "stateless": true,
      },
      "methods": Array [],
      "name": "App",
      "parentId": "8936d8c0447672f0ac71feaaaff306f8",
      "snippet": "const App = () => (<div>Hello</div>);",
      "start": 16,
      "superClass": null,
      "type": "class",
    },
    Object {
      "attributes": Array [],
      "base": "div",
      "end": 16,
      "id": "3d4d669d2bbd44aa9cada84ef5596d22",
      "name": "div",
      "parentId": "8936d8c0447672f0ac71feaaaff306f8",
      "snippet": "const App = () => (<div>Hello</div>);",
      "start": 16,
      "type": "jsx-usage",
    },
    Object {
      "end": 16,
      "id": "d377cfab7fe553f42333080ab8608951",
      "parentId": "8936d8c0447672f0ac71feaaaff306f8",
      "snippet": "const App = () => (<div>Hello</div>);",
      "start": 16,
      "text": "Hello",
      "type": "jsx-text",
    },
  ],
  "id": "8936d8c0447672f0ac71feaaaff306f8",
  "jsdoc": Array [],
  "mimeType": "text/javascript",
  "name": "hello",
  "nodeType": "File",
  "parentId": "1",
  "relativePath": "hello",
}
`;
